app/services/compute_attribute_merge.rb

```diff
@@ -0,0 +1,12 @@
+class ComputeAttributeMerge
+  def run(host, compute_attrs)
+    return if compute_attrs.nil?
+
+    host.compute_attributes ||= {}
+
+    vm_attrs = compute_attrs.vm_attrs
+    vm_attrs.deep_merge!(host.compute_attributes)
```

Comment:
Nitpick, you might want to avoid using `!` to not change the compute profile itself in memory, just run deep_merge and assign the result.